Makefile

```diff
@@ -1,96 +1,79 @@
-OUTPUT := .output
-TARGET := event_monitor_ebpf
+.PHONY: all
+all: build
+
+# environment:
+ARCH := $(shell uname -m)
 KERN_RELEASE := $(shell uname -r)
 KERN_SRC := /lib/modules/$(KERN_RELEASE)/build
-ARCH := $(shell uname -m | sed 's/x86_64/x86/')
-LLC ?= llc
-CLANG ?= clang
-LLVM_STRIP ?= llvm-strip
-BPF_C = tracee/${TARGET:=.c}
-BPF_OBJ = ${OUTPUT}/${TARGET:=.o}
-LIBBPF_SRC = $(abspath 3rdparty/libbpf/src)
-LIBBPF_OBJ = $(abspath $(OUTPUT)/libbpf.a)
-CFLAGS ?= -I$(OUTPUT)/usr/include/
+# inputs and outputs:
+OUT_DIR := dist
+BPF_SRC := tracee/tracee_ebpf.c 
+BPF_OBJ := $(OUT_DIR)/tracee_ebpf.o
+GO_SRC := $(shell find . -type f -name '*.go' ! -name '*_test.go')
+OUT_BIN := $(OUT_DIR)/tracee
+LIBBPF_SRC := 3rdparty/libbpf/src
+LIBBPF_HEADERS := $(OUT_DIR)/libbpf/usr/include
+LIBBPF_OBJ := $(OUT_DIR)/libbpf/libbpf.a
+OUT_DOCKER := tracee
+# tools:
+LLC := llc
+CLANG := clang
+LLVM_STRIP := llvm-strip
+# DOCKER := docker
+# GORELEASER := goreleaser
 
-SRC = $(shell find . -type f -name '*.go' ! -name '*_test.go' )
-ebpfProgramBase64 = $(shell base64 -w 0 tracee/event_monitor_ebpf.c)
+$(OUT_DIR):
+	mkdir -p $@
 
 .PHONY: build
-build: llvm-check $(BPF_OBJ) dist/tracee
-
-.PHONY: build-docker
-build-docker: clean
-	img=$$(docker build --target builder -q  .) && \
-	cnt=$$(docker create $$img) && \
-	docker cp $$cnt:/tracee/dist - | tar -xf - ; \
-	docker rm $$cnt ; docker rmi $$img
-
-dist/tracee: $(SRC) tracee/event_monitor_ebpf.c
-	GOOS=linux go build -v -o dist/tracee -ldflags "-X github.com/aquasecurity/tracee/tracee.ebpfProgramBase64Injected=$(ebpfProgramBase64)"
+build: $(OUT_BIN)
 
-.PHONY: test
-test:
-	go test -v ./...
-
-.PHONY: clean
-clean:
-	rm -rf dist || true
-	cd $(LIBBPF_SRC) && $(MAKE) clean;
-	rm -f $(BPF_OBJ)
-	rm -f *.ll
-	rm -rf .output
+$(OUT_BIN): $(BPF_OBJ) $(LIBBPF_OBJ) $(GO_SRC) | $(OUT_DIR)
```

Comment:
Tracee binary (OUT_BIN) doesn't depend on BPF_OBJ - they can be built independently

Reply:
yes, it was depended before when I embedded it in. will fix